staging/src/k8s.io/client-go/tools/auth/exec/types_test.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package exec
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	clientauthenticationv1beta1 "k8s.io/client-go/pkg/apis/clientauthentication/v1beta1"
+	clientcmdv1 "k8s.io/client-go/tools/clientcmd/api/v1"
+)
+
+// TestCluster ensures that clientauthenticationv1beta1.Cluster stays in sync with
+// clientcmdv1.Cluster.
+//
+// We want clientauthenticationv1beta1.Cluster to offer the same knobs as clientcmdv1.Cluster to
+// allow someone to connect to the kubernetes API. This test should fail if a new field is added to
+// one of the structs without updating the other.
+//
+// TODO: when clientauthenticationv1 is born, we should probably have a similar test.
```

Comment:
I'm all ears for suggestions on how we can make sure we don't forget to do this when we got to GA. I know we can add a TODO item somewhere in GitHub, but I would love to be able to add something in code.

Reply:
you can check if a ExecCredential is registered for v1 into the scheme created in staging/src/k8s.io/client-go/tools/auth/exec/exec.go and fail with a "TODO: add field fuzzing test" if so (can test it with a check for v1beta1 to make sure the check works)

Reply:
excellent. 56d9b8fda37